### PR TITLE
 Implementation of Update Cart endpoint - Pre-step 1

### PR DESCRIPTION
**Pull request by dumega**
# Description

Move the supports for Magento GiftCard from Model/Api/DiscountCodeValidation.php to Helper/Discount.php for common reuse.

Next PR : https://github.com/BoltApp/bolt-magento2/pull/851

Related task https://boltpay.atlassian.net/browse/M2P-64

Fixes: (link Jira ticket)

#changelog  Implementation of Update Cart endpoint - Pre-step 1-1

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Helper/Discount.php [hunk 1]
```diff
@@ -197,6 +197,16 @@ class Discount extends AbstractHelper
...
+    
+    /**
+     * @var ThirdPartyModuleFactory
+     */
+    protected $moduleGiftCardAccount;
+
+    /**
+     * @var ThirdPartyModuleFactory
+     */
+    protected $moduleGiftCardAccountHelper;
```
Comment (danac-gs) [inline Helper/Discount.php:209]: This seems unused.

**Comment by danac-gs** [comment]
@dumega you should probably fix / update unit tests

**Comment by dumega** [comment]
Unit tests are created, but there is an error caused by another issue https://github.com/BoltApp/bolt-magento2/pull/862

File: Helper/Discount.php [hunk 2]
```diff
@@ -1193,4 +1219,111 @@ public function applyMiravistRewardPoint($immutableQuote)
...
+    }
+    
+    /**
+     * @param string $type
+     * @return string
+     */
+    public function getBoltDiscountType($type)
+    {
+        switch ($type) {
+            case "by_fixed":
```
Comment (vitaliyreznikov) [inline Helper/Discount.php:1305]: We will change this constants in next PRs, correct?

**Pull request merged**